generators/generator-microsoft-bot-adaptive/generators/app/index.js

```diff
@@ -228,64 +146,83 @@ module.exports = class extends Generator {
     );
   }
 
-  _copySchemas() {
-    const botName = this.options.botName;
-    const directoryName = 'schemas';
-
-    this.fs.copyTpl(
-      this.templatePath(path.join('assets', directoryName)),
-      this.destinationPath(path.join(botName, directoryName)),
-      {
-        botName,
-      }
-    );
+  _writeJsPackageJson() {
+    const { botName, integration } = this.options;
+
+    const { label, start } = {
+      [integrations.functions]: { label: 'azure-functions' },
+      [integrations.webapp]: { label: 'restify', start: 'node index.js' },
+    }[integration];
+
+    const dependencies = {
+      [`botbuilder-runtime-integration-${label}`]: 'next',
+    };
+
+    this.fs.writeJSON(this.destinationPath(botName, 'package.json'), {
+      name: botName,
+      private: true,
+      scripts: {
+        start,
+      },
+      dependencies: Object.assign(
+        this.packageReferences.reduce(
+          (acc, { name, version }) => Object.assign(acc, { [name]: version }),
+          {}
+        ),
+        dependencies
+      ),
+    });
   }
 
   _writeApplicationSettings() {
-    const botName = this.options.botName;
-    const fileName = 'appsettings.json';
-
-    const filePath = path.join(
+    const {
+      applicationSettingsDirectory = '.',
       botName,
-      this.applicationSettingsDirectory
-        ? this.applicationSettingsDirectory
-        : '',
-      fileName
-    );
+      integration,
+      platform,
+    } = this.options;
 
     const appSettings = this.fs.readJSON(
-      this.templatePath(path.join('assets', fileName))
+      this.templatePath('assets', 'appsettings.json')
     );
 
     appSettings.luis.name = botName;
-    appSettings.runtime.command = `dotnet run --project ${botName}.csproj`;
-    appSettings.runtime.key = 'adaptive-runtime-dotnet-webapp';
 
-    for (const pluginDefinition of this.pluginDefinitions) {
-      appSettings.runtimeSettings.plugins.push(pluginDefinition);
-    }
+    appSettings.runtime.key = `adaptive-runtime-${platform}-${integration}`;
+    appSettings.runtime.command = {
+      [platforms.dotnet]: `dotnet run --project ${botName}.csproj`,
+      [platforms.js]: 'node index.js',
+    }[platform];
 
-    if (this.modifyApplicationSettings) {
-      this.modifyApplicationSettings(appSettings);
+    for (const { name, settingsPrefix } of this.packageReferences) {
```

Comment:
It's worth calling out - not every package will necessarily be a plugin. Some packages may contain declarative assets only and not need to be wired up as a plugin. I don't think we should necessarily assume they are always one and the same.

Reply:
Declarative assets still need to be registered with a resource explorer (see [Welcome package](https://github.com/microsoft/botframework-components/blob/b6a77a7a260663fb2c368af6b580aad34c9efc2c/packages/HelpAndCancel/index.js)).

Reply:
Seems like this may be a shortcoming of the JS SDK where declarative assets in `node_modules` aren't discovered. Something I'll work on this week, for now we will keep this and unroll it later.